src/common/runtime/helper/test_worker.ts

```diff
@@ -2,9 +2,16 @@ import { LogMessageWithStack } from '../../internal/logging/log_message.js';
 import { TransferredTestCaseResult, LiveTestCaseResult } from '../../internal/logging/result.js';
 import { TestCaseRecorder } from '../../internal/logging/test_case_recorder.js';
 import { TestQueryWithExpectation } from '../../internal/query/query.js';
+import { assert } from '../../util/util.js';
 
 import { CTSOptions, kDefaultCTSOptions } from './options.js';
 
+interface WorkerTestRunRequest {
+  rec: TestCaseRecorder;
```

Comment:
I'd like to use WorkerTestRunRequest as the actual object that we post to the worker so the worker can cast to it (in wrapTestGroupForWorker and test_worker-worker.ts)
though I see that because we have `this.ctsOptions` that might not work. Let me look into it a little.

Reply:
Ended up using it only in the postMessage/onmessage interface and not in makeRequestAndRecordResult. It's still nice though.